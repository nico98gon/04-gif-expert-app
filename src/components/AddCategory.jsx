import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, setinputValue ] = useState('');

    const onInputChange = ({ target }) => {
        setinputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault(); //Evita el refresh de la web
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim());
        setinputValue('');
    }

    return(
        <form onSubmit = { onSubmit }>
            <input
                type = "Text"
                placeholder = "Buscar gifs"
                value = { inputValue }
                onChange = { onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}