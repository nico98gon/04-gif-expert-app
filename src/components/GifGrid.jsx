import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
    
    const [images, setImages] = useState([]);

    const getImages = async() => {
        const newImages = await getGifs( category );
        setImages( newImages );
    }

    useEffect( () => {
        getImages();
    }, []);


    return (
        <>
            <h3>{ category }</h3>

            {
            // images.map( image => (
            //     <li key={image.id}>{ image.title }</li>
            // ))
            images.map( ({ id, title }) => (
                <li key={ id }>{ title }</li>
            ))
            }
        </>
    )
}
