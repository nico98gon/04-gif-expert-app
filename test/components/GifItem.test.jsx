import { GifItem } from "../../src/components/GifItem";
import { render, screen } from "@testing-library/react";

describe('Pruebas GifExpertApp', () => { 
    
    const title = 'Saitama';
    const url = "https://one-punch.com/saitama.jpg";
    
    test('should de hacer match con el snapshot', () => { 

        const { container } = render( <GifItem title="{ title }" url="{ url }" />)
        expect( container ).toMatchSnapshot();

    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => { 

        render( <GifItem title= { title } url= { url } />)
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        // screen.debug();
        const { src, alt} = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);

    });

    test('debe mostra el titulo en el componente', () => {
        render( <GifItem title= { title } url= { url } />)
        expect(screen.getByText( title )).toBeTruthy();
    })
    
});