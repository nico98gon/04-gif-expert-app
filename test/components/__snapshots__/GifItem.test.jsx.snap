// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifExpertApp should de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="{ title }"
      src="{ url }"
    />
    <p>
      { title }
    </p>
  </div>
</div>
`;
